Greg

gpric024 opened the pull request:
The test file is for testing the format of sentences. The load datasets python file demonstrates on how to get the wmt data (it does not actually download or do anything)

gpric024 committed:
Uploading original results notebook

gpric024 committed:
adding the injected errors process

gpric024 committed:
adding class files and dicts

gpric024 committed:
removing old files

gpric024 committed:
how to get the datasets

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+                self.data = self.data[0].tolist()
+            else:
+                raise Exception("Invalid file type")
+        elif data is not None:
```
When will this happen? Is it an expected use case to initialize the DataLoader with a list of sentences?

Shaomei replied:
ah, I see, this is for the deepcopy..

gpric024 replied:
there is probably a more efficient way of doing this

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+
+    def parse_txt(self, path):
+        output = []
+        with open(path, "r") as f:
```
```suggestion
        with open(path, "r", encoding="utf-8") as f:
```

Shaomei replied:
It is often a good idea to specify the unicode encoding, unless you are parsing ascii/binary files.

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+        return sentence
+
+    def save_as_txt(self, path):
+        with open(path, "w") as f:
```
```suggestion
        with open(path, "w", encoding="utf-8") as f:
```

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+    
+    def save_as_csv(self, path):
+        df = pd.DataFrame(self.data)
+        df.to_csv(path, index=False, header=False)
```
```suggestion
        df.to_csv(path, index=False, header=False, encoding='utf-8')
```

Shaomei replied:
Although utf-8 is the default encoding for python 3, it is not the case for python 2. Better to be explicit here.

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+    def get_data(self):
+        return self.data
+
+    def create_deepcopy(self):
```
I guess this is the use case for the line 19, as I commented above. But it is unclear what the deepcopy is for? Is it for the cleaned up data? Some comments here can help clarify.

Shaomei replied:
Okay, I see where this deepcopy is used now. Ignore this now. But perhaps there can be an cleaner way to do this.

gpric024 replied:
There is definitely a cleaner way of doing it. I will see what I can think of

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+import random
+import DataLoader #Custom class
```
nit: I usually put the custom class below all the system classes.

gpric024 replied:
Got it. Will try to remember

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,76 @@
+import re
+import pandas as pd
+import copy
+class DataLoader:
```
Usually a good practice to use docstring here to explain the class and its usage. E.g. 

```
"""Loader for benchmarking datasets

(Longer explanation about the class)

Attributes:
    dataset_name:
    data:
"""
```

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+            #create deep copy of the data
+            if p1 == 0:
+                temp_load, results = self.injection_runner(self.load.create_deepcopy(), p1, p2)
+                df_swap_results.loc[len(df_swap_results)] = {"dataset":"wmt14_en", "p1":0, "p2":0, "words_swapped":results[0], "letters_swapped":results[1], "words_changed":results[2], "sentences_changed":results[3]}
```
nit: line too long, consider breaking into multiple lines.

gpric024 replied:
Done

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+
+            #update the p1 and p2
+            p1 += step_size
+            p2 += step_size
```
Why are we incrementing p1 and p2 at the same time? Should this entire while loop be replaced with two nested for loops? Like:

```
for p1 in range(p_start, p_end, step_size):
    for p2 in range(p_start, p_end, step_size):
         temp_load, results = self.injection_runner(self.load.create_deepcopy(), p1, p2)
         df_swap_results.loc[len(df_swap_results)] = {"dataset":self.load.dataset_name,
                                                      "p1":p1,
                                                      "p2":p2,
                                                      "words_swapped":results[0],
                                                      "letters_swapped":results[1],
                                                      "words_changed":results[2],
                                                      "sentences_changed":results[3]}
         self.saver(temp_load, save_path, p1, p2, save_format)
```

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+        return confusing_letters_dict
+    
+    def injection_swap(self, p_start=0, p_end=1, step_size=0.1, save_path="", save_format="both"):
+        df_swap_results = pd.DataFrame(columns=["dataset","p1", "p2", "words_swapped", "letters_swapped", "words_changed", "sentences_changed"])
```
Consider changing p1, p2 to more meaningful names, such as p_letter, p_homophone.

gpric024 replied:
Done

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+        #we also need to keep track of the amount of words that were changed
+        words_changed = 0
+        for i in range(len(words)):
+            #if the word has any punctuation or capitalization, we skip it
```
We probably should not skip capitalized words or words with punctuation - even if it is harder to do the homophone confusion, we can still do the letter swapping, right?

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+                #we need to check if the word has any homophones
+                if len(self.homophones_dict[word]) > 0:
+                    #we need to check if we will swap the word with a homophone with probability p1
+                    
```
nit: remove empty line.

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+            word = words[i].lower().strip('".,?!:;()')
+            #create a copy of the word so we can check if it was changed
+            word_copy = word
+            #we need to check if the word is in the homophones dict
```
nit: remove "we need to" in the comments.

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,167 @@
+                        #we need to update the amount of words that were swapped
+                        words_swapped += 1
+                        continue
+            p_letter = p2 #lower the probability of swapping a letter with a confusing letter each time we swap a letter
```
Move this comment to line 159

Shaomei reviewed (commented):
Hi Greg,

I did a quick pass and shared some comments. Overall it looks great - much better than the notebook, I am very glad to see the code modularized and well-organized. Keep up the good work!

Shaomei reviewed (commented):
Hi Greg,

I did a quick pass and shared some comments. Overall it looks great - much better than the notebook, I am very glad to see the code modularized and well-organized. Keep up the good work!

gpric024 committed:
Update DataLoader.py

adding encoding commit suggestion

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

Same as last commit

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

Same as last two commits

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

added doc string

gpric024 committed:
Update DyslexiaInjector.py

Updated with Shaomei's comments in mind, still need to address capitalization and nested loops for p_homophone and p_letter

gpric024 committed:
Update DyslexiaInjector.py

Modified code so we now include capitalized words and words with punctuation in our swapping corpus

gpric024 committed:
Update DyslexiaInjector.py

Just updated some comments

gpric024 committed:
Updates

Made required changed to dyslexia injector and data loader. Bug fixes and slightly optimized.

gpric024 commented:
Fixes will be present in next pull request
